### PR TITLE
Add OpenFF Docs template

### PR DESCRIPTION
**Pull request by Yoshanuikabundi**
Changes made in this Pull Request:
 - MyST setup
 - OpenFF sphinx theme configuration
 - Intersphinx setup with all existing OpenFF tools
 - Move docs environment to devtools/conda-envs
 - Automatic API docs documentation
 - Changes to docstrings and `__all__` to demonstrate API docs setup

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG.md updated?
 - [ ] AUTHORS.md updated?

### CODE REVIEW

File: {{cookiecutter.repo_name}}/docs/make.bat [hunk 1]
```diff
@@ -9,7 +9,7 @@ if "%SPHINXBUILD%" == "" (
 )
 set SOURCEDIR=.
 set BUILDDIR=_build
-set SPHINXPROJ={{cookiecutter.repo_name}}
+set SPHINXPROJ=openff-nagl
```
Comment (lilyminium) [inline {{cookiecutter.repo_name}}/docs/make.bat:12]: Is this change on purpose?

**Review by lilyminium — CHANGES_REQUESTED** [review]
Looking good, just a question about a nagl mention.

I'm not sure if there's an easy way to test out the docs -- the MDAnalysis cookiecutter pushes to a "cookie" repo as part of CI, which I could re-implement if that would be useful.

File: {{cookiecutter.repo_name}}/docs/make.bat (hunk 1 above)
Comment (Yoshanuikabundi) [reply {{cookiecutter.repo_name}}/docs/make.bat]: Nope! Fixed in 7d92590

**Comment by Yoshanuikabundi** [comment]
Could do the GH Pages thing and push to a branch, and then have RTD render it as a version in the cookiecutter docs? I haven't thought much about testing, since we're only going to use it internally I figure if there's an issue that can be detected by tests then all its gonna do is cause a cookie to fail when we try to create it. Not the end of the world.

This wouldn't be for this PR, but would it be helpful if I tried to get the package structure set up like OpenFF projects - ie, `openff.packagename` rather than `openff-packagename`?

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
> This wouldn't be for this PR, but would it be helpful if I tried to get the package structure set up like OpenFF projects - ie, openff.packagename rather than openff-packagename?

Yes, that would be fantastic!

**Comment by lilyminium** [comment]
LGTM, thank you!

**Pull request merged**